### ISSUE TITLE
Oracle: no violations found when renaming constraints and indexes

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#7044]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

FIles with ALTER TABLE with constraint renames are not processed.

### Expected Behaviour

The file should be processed as expected.

### Observed Behaviour

The files are not processed.

### How to reproduce

1. Create a migration file (e.g., /workspace/migrations/example_migration.sql) as follows:

```sql
alTeR tablE mY_TABle renaMe CONSTRAINT my_conSTRAINT TO CTX_my_conSTRaint;
alTER index my_INdex RENAME to PK_my_iNDEx;
```

2. Configure sqlfluff as needed (see configuration section).
3. Install and run sqlfluff on that file:

```sh
pip install --no-cache-dir sqlfluff==3.4.2
sqlfluff fix /workspace/migrations/example_migration.sql
```

The response from sqlfluff is below. Note that the file parsing works:

```sh
==== finding fixable violations ====
  [1 templating/parsing errors found]    <-- This line does not appear if you remove the ALTER INDEX
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

4. Replace the content of the file with the following:

```sql
alTeR tablE mY_TABle ADD CONSTRAINT my_conStRainT FOREIGN KEY (some_col) REFERENCES another_TaBle (id) ON delete CASCADE;
```

5. Run sqlfluff again:

```sh
sqlfluff fix /workspace/migrations/example_migration.sql

==== finding fixable violations ====
== [/workspace/migrations/example_migration.sql] FAIL
L:   1 | P:   1 | CP01 | Keywords must be upper case. [capitalisation.keywords]
L:   1 | P:   7 | CP01 | Keywords must be upper case. [capitalisation.keywords]
L:   1 | P:  13 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   1 | P:  37 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   1 | P:  85 | CP02 | Unquoted identifiers must be lower case.
                       | [capitalisation.identifiers]
L:   1 | P: 107 | CP01 | Keywords must be upper case. [capitalisation.keywords]
== [/workspace/migrations/example_migration.sql] FIXED
6 fixable linting violations found
```

This time it works as expected!

### Dialect

oracle

### Version

3.4.2

### Configuration

I tested it with the following .sqlfluff configuration file in /workspace:

```yaml
[sqlfluff]
templater = jinja
dialect = oracle
ignore = templating
large_file_skip_byte_limit = 0
max_line_length = 0

[sqlfluff:layout:type:alias_expression]
spacing_before = align
align_within = select_clause
spacing_after = touch

[sqlfluff:indentation]
tab_space_size = 2
indent_unit = space
indented_joins = false
indented_using_on = true
allow_implicit_indents = true
indented_on_contents = false
indented_ctes = false

[sqlfluff:rules:aliasing.table]
aliasing.table = explicit

[sqlfluff:rules:aliasing.column]
aliasing.column = explicit

[sqlfluff:rules:aliasing.expression]
allow_scalar = True

[sqlfluff:rules:ambiguous.join]
fully_qualify_join_types = inner

[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = consistent

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
unquoted_identifiers_policy = all

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:jinja.padding]
single_space = true

[sqlfluff:rules:layout.spacing]
no_trailing_whitespace = true
extra_whitespace = false

[sqlfluff:rules:layout.commas]
line_position = trailing

[sqlfluff:rules:layout.functions]
no_space_after_function_name = true

[sqlfluff:rules:layout.select_targets]
wildcard_policy = single

[sqlfluff:rules:layout.set_operators]
set_operator_on_new_line = ['UNION', 'UNION ALL']

[sqlfluff:rules:structure.nested_case]

[sqlfluff:rules:structure.subquery]
forbid_subquery_in = join
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by saulotoledo** [comment]
@WittierDinosaur, @tunetheweb, PRs #7045 and #7046 should fix this issue.

**Comment by keraion** [comment]
Closing per completed linked items
